pkg/agent/api/logging.go

```diff
@@ -5,116 +5,139 @@ package api
+		}
 		if err != nil {
-			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			args = append(args, slog.Any("error", err))
+			lm.logger.Warn("Publish message failed to complete successfully.", args...)
 			return
 		}
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info("Publish message completed successfully.", args...)
 	}(time.Now())
 
 	return lm.svc.Publish(topic, payload)
 }
 
 func (lm loggingMiddleware) Execute(uuid, cmd string) (str string, err error) {
 	defer func(begin time.Time) {
-		message := fmt.Sprintf("Method exec for uuid %s and cmd %s took %s to complete", uuid, cmd, time.Since(begin))
+		args := []any{
+			slog.String("duration", time.Since(begin).String()),
+			slog.String("uuid", uuid),
+			slog.String("cmd", cmd),
+		}
 		if err != nil {
-			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			args = append(args, slog.Any("error", err))
+			lm.logger.Warn("Execute command failed to complete successfully.", args...)
 			return
 		}
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info("Execute command completed successfully.", args...)
 	}(time.Now())
 
 	return lm.svc.Execute(uuid, cmd)
 }
 
 func (lm loggingMiddleware) Control(uuid, cmd string) (err error) {
 	defer func(begin time.Time) {
-		message := fmt.Sprintf("Method control for uuid %s and cmd %s took %s to complete", uuid, cmd, time.Since(begin))
+		args := []any{
+			slog.String("duration", time.Since(begin).String()),
+			slog.String("uuid", uuid),
+			slog.String("cmd", cmd),
+		}
 		if err != nil {
-			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			args = append(args, slog.Any("error", err))
+			lm.logger.Warn("Control command failed to complete successfully.", args...)
 			return
 		}
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info("Control command completed successfully.", args...)
 	}(time.Now())
 
 	return lm.svc.Control(uuid, cmd)
 }
 
 func (lm loggingMiddleware) AddConfig(c agent.Config) (err error) {
 	defer func(begin time.Time) {
-		message := fmt.Sprintf("Method add_config took %s to complete", time.Since(begin))
+		duration := slog.String("duration", time.Since(begin).String())
 		if err != nil {
-			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			lm.logger.Warn("AddConfig failed to complete successfully.", duration, slog.Any("error", err))
 			return
 		}
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info("AddConfig completed successfully.", duration)
 	}(time.Now())
 
 	return lm.svc.AddConfig(c)
 }
 
 func (lm loggingMiddleware) Config() agent.Config {
 	defer func(begin time.Time) {
-		message := fmt.Sprintf("Method config took %s to complete", time.Since(begin))
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info("Config completed successfully.", slog.String("duration", time.Since(begin).String()))
```

Comment:
```suggestion
		lm.logger.Info("Retrieve config  completed successfully.", slog.String("duration", time.Since(begin).String()))
```

Look at what the functions are doing then log the correct message

Reply:
This has been changed recently in Magistrala to provide more human-friendly logs.